Marked the xmlrpc_api_url method as deprecated.

zinigor opened the pull request:
This changes the way the Actions class in the Sync package uses the `xmlrpc_api_url` method and adds the deprecation function call to the old one.

<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

Fixes #14048 

#### Changes proposed in this Pull Request:
* Changes the way we use the method in Actions.
* Adds a call do `_deprecated_function` in the main Jetpack class.

#### Testing instructions:
* Try connecting a new JN site, add something like a post, a new user, etc.
* Make sure the data is synchronized either using the cache site browser, or Calypso.

#### Proposed changelog entry for your changes:
* N/A

zinigor committed:
Marked the xmlrpc_api_url method as deprecated.

This changes the way the Actions class in the Sync package uses the `xmlrpc_api_url` method and adds the deprecation function call to the old one.

jeherve commented on class.jetpack.php:
```diff
@@ -5130,6 +5130,7 @@ public function alert_auto_ssl_fail() {
 	 * @return string
 	 */
 	public static function xmlrpc_api_url() {
+		_deprecated_function( __METHOD__, 'jetpack-8.0' );
```
Should we expand on this here, something like this?

```suggestion
		_deprecated_function( __METHOD__, 'jetpack-8.0', 'Automattic\\Jetpack\\Connection\\Manager::xmlrpc_api_url()' );
```

kbrown9 reviewed (dismissed):
I tested this branch by connecting Jetpack and creating a new post. The new post was visible in Calypso as expected.

zinigor committed:
Added an extended deprecation argument.

Props to @jeherve for the suggestion, I have forgotten about the third argument.

Co-Authored-By: Jeremy Herve <jeremy@jeremy.hu>

zinigor commented:
Whoops, sorry @kbrown9 , your review got dismissed by a suggestion commit. Can you approve again, please?
